### ISSUE TITLE
FB Information to be used when a person figures as a relative

### ISSUE THREAD

**Issue by tuurma** [eXistSolutions/LGPN#296]
> RC: Information given in the final bracket does not appear when a person figures as a relative. So people with a double name connected by some formula (e.g. ὁ καὶ) lose that information as a relative. This can result in erroneous accentuation appearing in the FB (e.g. Σεανιος 1 where the son appears as Ἰταλός Ταμαλατος instead of Ἰταλὸς ὁ Ταμαλλατος). Even more serious is the loss of the Roman praenomen and nomen which only appear in the FB and the section for Roman names. In all previous vols. the Roman elements of the name are given when the person occurs as a relative in the FB. This may be a problem connected with the way we structured the input form, but could you see if there is any way of getting around it?

> MT: If I understand right, the relative name should be using its own FB formulation (from the manual input), if available? 

![image](https://user-images.githubusercontent.com/449468/104304031-245b7580-54cb-11eb-8639-d7e5412733da.png)

I can do this, just concerned that sometimes FB contains additional info which may look odd. We could potentially divide the FB field into two - for the name part and "the rest" to avoid this issue. Also please note that in your example is entered in FB as Ἰτ(α)λὸς Ταμα[λα]τος, also missing the connecting  ὁ, also it is not noted in the 'linking' field of either name. We're dealing with a combination of issues, some in my algorithms to determine the name, but some in the data entry.

_Originally posted by @tuurma in https://github.com/eXistSolutions/LGPN/issues/292#issuecomment-758571897_

**Comment by tuurma** [comment]
@RichardLGPN

Current state of the algorithm: 
1. uses the name from FB where available, 
2. otherwise follows to construct the name from listed names, adding the linking formula where available

Therefore the output for Σεανιος 1 is:

![image](https://user-images.githubusercontent.com/449468/106497814-d086ef80-64be-11eb-8337-47c57c752ebd.png)

In my local version, for test purposes I have removed the FB and added ὁ in the linking field for Ταμαλατος, thus the output changes to:

![image](https://user-images.githubusercontent.com/449468/106497963-ff9d6100-64be-11eb-9609-5720d135ac69.png)

I hope this covers most of the cases we'll be dealing with.

**Comment by RichardLGPN** [comment]
I don't think this is going to work quite so easily. I was working on some entries for Persians yesterday, some of whose attestations in non-Greek sources are recorded, and the whole list of variations gets cited for the person as relative. Just have a look at some of the people called Δαρεῖος and you will see what I mean.
________________________________

**Comment by tuurma** [comment]
Indeed, that's exactly what I was afraid of happening. So, our options are:

* divide the FB field into two, the "name" part and the "rest"; unfortunately this would require edits to the names affected
* try to reconstruct the composite name from individual nym entries; as far as I remember main problem here were Roman elements of the name which was why we had them entered in the FB in the first place

What do you think?

**Comment by tuurma** [comment]
There are almost 18 thousand entries in V6 with FB filled, and 2650 with a Roman name component

**Comment by RichardLGPN** [comment]
I need to think a bit more about this and the possible complications that might arise. I'll get back to you as soon as possible.

Richard


________________________________

**Comment by RichardLGPN** [comment]
In deciding how to deal with this problem we have to take the following into account


  1.  When Roman names are in Latin (e.g. M. Aur. Severus), we do not translate the elements into Greek. So this person appears as a relative as plain Σεουῆρος.
  2.  It is normal for a person with Roman names to be entered in the final bracket with the cognomen abbreviated to the initial letter – unless the cognomen has a variant spelling. So Μ. Αὐρ. Σεουῆρος becomes Μ. Αὐρ. Σ. This makes it useless for the purpose of generating the name in full as a relation.
  3.  For Roman names, the information entered in the Roman Names section is of no use as it is entered sometimes in Greek, sometimes in Latin, as well as sometimes fully expanded (e.g. from G. to Gaius or Γ. to Γἀιος.
  4.  Where names, whether of Roman type or multiple, are attested with a variant spelling, that is the form in which it appears in the FB, whereas it should appear with normalized spelling as a relation.

Given all this, only your first option seems workable, and whatever the solution I can't see any way of avoiding a good deal of manual editing. Actually, this being so, I wonder if a more radical solution is not required in the form of adding a new box in the Names field where the form of the name should be entered as it should appear as a relation. This would only be necessary in those cases which concern us, the default being the name that occurs in the Relationships field. Whatever is decided, it should be a choice that takes into account not just the needs of the current volume but all future use of this database by us and, potentially, other partners doing updates to previous volumes as has been planned for LGPN I.

To get an idea of the scale of the task, can you get totals for the people a) who have the Roman names but only when attested in Greek and b) who have double names, ideally only those where a connecting formula is used.
________________________________

**Comment by tuurma** [comment]
* Roman names attested in Greek (e.g. Τιβέριος Κλαύδιος below): 2651 in V6, 2699 total

```xml
<nym ana="greek" corresp="s1" n="1" type="Roman">
                            <persName type="praenomen">Τιβέριος</persName>
                            <persName type="nomen_gentile">Κλαύδιος</persName>
                           
</nym>
```

* double names: 3247 in V6, 3314 total
* double names with a connecting formula: only 118 in V6, 131 total

**Comment by tuurma** [comment]
Dear Richard, shall I prepare the ground and add the 'Name as a relation' box (better label welcome)? It's a small adjustment to the editing form but also requires running a query across the database to add such a field to all existing entries, therefore I'd rather do it when you all are not working, e.g. late afternoon/early evening.

**Comment by RichardLGPN** [comment]
Would 5pm (UK time) suit you? How long do you estimate it will take? I will let Michael know.

Richard
________________________________

**Comment by tuurma** [comment]
Yes, perfect, thank you. Update itself will not be long, 15mins or so, but the reindexing needs to run afterwards (during which you can work, just searches will be unreliable until it finishes)

**Comment by tuurma** [comment]
After yesterday's update, the editing form and the modules to generate book proofs are adjusted to use the name-as-relative field, if present, and automatically constructed name otherwise.

Shall I help somehow with identifying the names that need manual adjustment? Entries with  Roman name components seem like first candidates for review.

**Comment by RichardLGPN** [comment]
Yes, if possible, I shall need lists of people with Roman names and those with double names.

Richard
________________________________

**Comment by tuurma** [comment]
Dear Richard, you'll find a list of people with double names here: http://clas-lgpn4.classics.ox.ac.uk:8080/exist/apps/lgpn-editor/modules/tools/doubleNames.xql 

Second column *Generated name* lists the combined name form I could generate for use as a name of the relative. At the moment I list name components in the order they are entered, but could easily make Roman components appear in front, if that helps.

There are some obvious cases where two names have been entered but one is empty (e.g. 20 Μοαιαρος) or linking formula is attached to the wrong part (e.g. 70 ὁ καὶ Αὐγουστιανός Πάρις Οὔλπιος). These would only require fixes to the "regular" parts of the form.

If you manually edit the entry and fill in the Name-as-relative field, it would be used in the book generation instead of automatically generated name.

Please let me know if I could add or rearrange something to make your work easier.

**Comment by RichardLGPN** [comment]
Many thanks for this. I will get started on it next week.

Richard
________________________________

**Comment by RichardLGPN** [comment]
Dear Magdalena,

I have finally finished going through the list you sent, a long and tedious job but one that was necessary. In the course of it I identified a number of duplicates which might otherwise have passed undetected, so that was a bonus.

What I now need is a list of all those people who have non-Greek renditions of their names, sometimes in addition to a Greek version. So everyone with Lat., Syr., Heb., Aram., Palm., Nab., Phoen., Arab., Saf., Bab., Pers., Parth., Elam., Arm., Lyc., Armen., Prakrit.

By the way, we need to add Armenian and Prakrit to the drop-down list of languages.

You may remember I mentioned the issue of differentiating between Syria, Antioch? and Syria?, Antioch. Is the solution simply to authorize each separately?

A parallel problem are those author references, as in the case of Procopius, where sometimes the ref. is followed by a simple book and chapter number (e.g. Procop. ii 12), sometimes by a book title which introduces a comma after the author name (e.g. Procop., +Arc. xxii 13). I think you may have suggested a solution to this some time ago which was never implemented and I can no longer lay my hands on.

Spring seems to have arrived - bright and warm sunny days, cold frosty nights. Are you still locked down?

best wishes,

Richard


________________________________

**Comment by RichardLGPN** [comment]
Could I also you for a file for all those people from Persia arranged by name - not just a list of names but the full details

I also need a similar file, this time arranged by place, for all entries where people are attested on coins.

I could perhaps do the first myself, rather slowly, but can see no way of doing the second.

best wishes,

Richard
________________________________

**Comment by tuurma** [comment]
Dear Richard,

I'm preparing the lists you asked for and meanwhile adding the languages. 
Still thinking about the Syria?, Antioch case for which I don't see a simple solution.

For Armenian, I'm assuming the Classical Armenian, ISO language code `xcl` https://iso639-3.sil.org/code/xcl
There are numerous entries for Prakrit (pka - Ardhamāgadhī Prākrit, pmh Māhārāṣṭri Prākrit, psu Sauraseni Prākrit) but also a collective `pra` https://iso639-3.sil.org/code/pra, so unless you tell me otherwise, I'll register it under the collective language name.

**Comment by RichardLGPN** [comment]
Thanks. As far as those two languages are concerned, it is indeed Classical Armenian. But for practical purposes we just need an indication of the basic language for the drop-down list for the Attested Form section.

Richard
________________________________

**Comment by tuurma** [comment]
People attested on coins, arranged by place: http://clas-lgpn4.classics.ox.ac.uk:8080/exist/apps/lgpn-editor/templates/subpages/coins.html

**Comment by tuurma** [comment]
Would such a display be sufficient for the list of people with names in non-Greek renditions? I'm splitting the lists by language because it's over 20 thousand names in total /o\

Where one of the name forms is attested in Greek, I skip the language marker (e.g. row 4 below)

![image](https://user-images.githubusercontent.com/449468/109498094-47e67980-7a93-11eb-9f0f-57b8dca8c936.png)

**Comment by tuurma** [comment]
I have added the link to the non-Greek renditions list also in the menu http://clas-lgpn4.classics.ox.ac.uk:8080/exist/apps/lgpn-editor/templates/subpages/nonGreek.html?la=sux

Initially, people with Babylonian renditions are displayed. Clicking on any of the language links listed at the top of the page will display a list of people for that language. In the table language codes are used, not full language names.

![image](https://user-images.githubusercontent.com/449468/109500135-1ae79600-7a96-11eb-89de-3b0a6580bd36.png)

**Comment by tuurma** [comment]
Please, be patient, list for `syr` (Syriac) takes quite a long time to load, I didn't check all others

![image](https://user-images.githubusercontent.com/449468/109500489-9c3f2880-7a96-11eb-8c7f-20dae9e106d6.png)
